src/instructlab/feature_flags.py

```diff
@@ -0,0 +1,32 @@
+import os
+from enum import StrEnum
+
+class FeatureScopes(StrEnum):
+    Default = 'Default'
+    DevPreviewNoUpgrade = 'DevPreviewNoUpgrade',
+    TechPreviewNoUpgrade = 'TechPreviewNoUpgrade',
+    CustomNoUpgrade = 'TechPreviewNoUpgrade'
+
+
+class GatedFeatures(StrEnum):
+    RAG = 'RAG'
+
+
+class FeatureGating:
+    env_var_name = 'ILAB_FEATURE_SCOPE'
+    _feature_scopes = {
+        GatedFeatures.RAG: FeatureScopes.DevPreviewNoUpgrade
+    }
+
+    @staticmethod
+    def available_scopes():
+        return [f.value for f in FeatureScopes]
+
+    @staticmethod
```

Comment:
just out of curiosity: why a static method instead of a function, here and in the rest of this module? (I'm not against it, but I see it's not widely adopted pattern in the project)

Reply:
In order to avoid having in instantiate a `FeatureGates` object.

Reply:
if we only use functions, the class it's not even needed. anyway, it's fine (I like it, it's just rarely used in the codebase)